src/Features/Core/Portable/Completion/Providers/TypeImportCompletionItem.cs

```diff
@@ -29,7 +29,8 @@ public static CompletionItem Create(INamedTypeSymbol typeSymbol, string containi
 
             // Hack: add tildes (ASCII: 126) to name and namespace as sort text:
             // 1. '~' before type name makes import items show after in-scope items
-            // 2. '~' before namespace makes types with identical type name but from different namespace all show up in the list
+            // 2. ' ' before namespace makes types with identical type name but from different namespace all show up in the list,
+            //    it also makes sure type with shorter name shows first, e.g. 'SomeType` before 'SomeTypeWithLongerName'.  
```

Comment:
Do we not have any tests that assert the relative ordering of two items? Don't have the test directly assert the sort text of course, but I'd expect we'd recreate this and at least say "this item appears before that item"?

Reply:
I will add a test for this

Reply:
We have `AssertItemsInOrder` but you have to give it the whole list. Maybe that thing could take a flag?

Reply:
Test added